userspace/libsinsp/filterchecks.cpp

```diff
@@ -1440,6 +1545,144 @@ bool sinsp_filter_check_fd::compare_port(sinsp_evt *evt)
 	return false;
 }
 
+bool sinsp_filter_check_fd::compare_domain(sinsp_evt *evt)
+{
+	if(!extract_fd(evt))
+	{
+		return false;
+	}
+
+	if(m_fdinfo != NULL)
+	{
+		scap_fd_type evt_type = m_fdinfo->m_type;
+		if(evt_type != SCAP_FD_IPV4_SOCK &&
+		   evt_type != SCAP_FD_IPV6_SOCK)
+		{
+			return false;
+		}
+
+		if(m_fdinfo->is_role_none())
+		{
+			return false;
+		}
+
+		uint32_t *addr;
+		if(m_field_id == TYPE_CLIENTIP_NAME)
+		{
+			if(evt_type == SCAP_FD_IPV4_SOCK)
+			{
+				addr = &m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_sip;
+			}
+			else
+			{
+				addr = &m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_sip.m_b[0];
+			}
+		}
+		else if(m_field_id == TYPE_SERVERIP_NAME)
+		{
+			if(evt_type == SCAP_FD_IPV4_SOCK)
+			{
+				addr = &m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_dip;
+			}
+			else
+			{
+				addr = &m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_dip.m_b[0];
+			}
+		}
+		else
+		{
+			bool is_local;
+			if(evt_type == SCAP_FD_IPV4_SOCK)
+			{
+				is_local = m_inspector->get_ifaddr_list()->is_ipv4addr_in_local_machine(m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_sip, m_tinfo);
+			}
+			else
+			{
+				is_local = m_inspector->get_ifaddr_list()->is_ipv6addr_in_local_machine(m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_sip, m_tinfo);
+			}
+
+			if(is_local)
+			{
+				if(m_field_id == TYPE_LIP_NAME)
+				{
+					if(evt_type == SCAP_FD_IPV4_SOCK)
+					{
+						addr = &m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_sip;
+					}
+					else
+					{
+						addr = &m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_sip.m_b[0];
+					}
+				}
+				else
+				{
+					if(evt_type == SCAP_FD_IPV4_SOCK)
+					{
+						addr = &m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_dip;
+					}
+					else
+					{
+						addr = &m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_dip.m_b[0];
+					}
+				}
+			}
+			else
+			{
+				if(m_field_id == TYPE_LIP_NAME)
+				{
+					if(evt_type == SCAP_FD_IPV4_SOCK)
+					{
+						addr = &m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_dip;
+					}
+					else
+					{
+						addr = &m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_dip.m_b[0];
+					}
+				}
+				else
+				{
+					if(evt_type == SCAP_FD_IPV4_SOCK)
+					{
+						addr = &m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_sip;
+					}
+					else
+					{
+						addr = &m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_sip.m_b[0];
+					}
+				}
+			}
+		}
+
+		uint64_t ts = evt->get_ts();
+
+		if(m_cmpop == CO_IN)
+		{
+			for (uint16_t i=0; i < m_val_storages.size(); i++)
```

Comment:
You might want to rely on the base class version of flt_compare here. For the _NAME filterchecks, you should just be able to extract them as a string and then do the normal comparison operators. This is especially important for IN as you can do a set membership test instead of looping over the values and comparing each value.

Reply:
The resolution goes from the domain name to the IP, not the other way around, so I have to call `sinsp_dns_manager::match()` at least once for every domain name in the list.
But, as optimization, after the domain name has been resolved I should be able to do what you're proposing because then `sinsp_dns_manager::name_of` will work.
I'll do that and then, if needed, we can talk more about it.

Reply:
Ah right, I see. The names are used as arguments to the matching function, not the values.